File: jscomp/test/mock_mt.ml
type eq = Mt.eq =
  | Eq : 'a * 'a -> eq
  | Neq : 'a * 'a -> eq
  | StrictEq : 'a * 'a -> eq
  | StrictNeq : 'a * 'a -> eq
  | Ok : bool -> eq
  | Approx : float * float -> eq
  | ApproxThreshold : float * float * float -> eq
  | ThrowAny : (unit -> unit) -> eq
  | Fail : unit -> eq
  | FailWith : string -> eq

type pair_suites = (string * (unit -> eq)) list

let from_pair_suites (name : string) (suites : pair_suites) =
  Js.log (name, "testing") ;
  List.iter
    (fun (name, code) ->
      match code () with
      | Eq (a, b) -> Js.log (name, a, "eq?", b)
      | Neq (a, b) -> Js.log (name, a, "neq?", b)
      | StrictEq (a, b) -> Js.log (name, a, "strict_eq?", b)
      | StrictNeq (a, b) -> Js.log (name, a, "strict_neq?", b)
      | Approx (a, b) -> Js.log (name, a, "~", b)
      | ApproxThreshold (t, a, b) -> Js.log (name, a, "~", b, " (", t, ")")
      | ThrowAny fn -> ()
      | Fail _ -> Js.log "failed"
      | FailWith msg -> Js.log ("failed: " ^ msg)
      | Ok a -> Js.log (name, a, "ok?"))
    suites

File: jscomp/test/mock_mt.js
'use strict';

var List = require("../../lib/js/list.js");
var Curry = require("../../lib/js/curry.js");

function from_pair_suites(name, suites) {
  console.log([
        name,
        "testing"
      ]);
  List.iter((function (param) {
          var name = param[0];
          var fn = Curry._1(param[1], undefined);
          switch (fn.TAG) {
            case "Eq" :
                console.log([
                      name,
                      fn._0,
                      "eq?",
                      fn._1
                    ]);
                return ;
            case "Neq" :
                console.log([
                      name,
                      fn._0,
                      "neq?",
                      fn._1
                    ]);
                return ;
            case "StrictEq" :
                console.log([
                      name,
                      fn._0,
                      "strict_eq?",
                      fn._1
                    ]);
                return ;
            case "StrictNeq" :
                console.log([
                      name,
                      fn._0,
                      "strict_neq?",
                      fn._1
                    ]);
                return ;
            case "Ok" :
                console.log([
                      name,
                      fn._0,
                      "ok?"
                    ]);
                return ;
            case "Approx" :
                console.log([
                      name,
                      fn._0,
                      "~",
                      fn._1
                    ]);
                return ;
            case "ApproxThreshold" :
                console.log([
                      name,
                      fn._1,
                      "~",
                      fn._2,
                      " (",
                      fn._0,
                      ")"
                    ]);
                return ;
            case "ThrowAny" :
                return ;
            case "Fail" :
                console.log("failed");
                return ;
            case "FailWith" :
                console.log("failed: " + fn._0);
                return ;
            
          }
        }), suites);
}

exports.from_pair_suites = from_pair_suites;
/* No side effect */
